src/server/scripts/DragonIsles/AberrusTheShadowedCrucible/boss_kazzara_the_hellforged.cpp

```diff
@@ -93,10 +278,492 @@ struct boss_kazzara_the_hellforged : public BossAI
     void UpdateAI(uint32 diff) override
     {
         scheduler.Update(diff);
+
+        if (!UpdateVictim())
+            return;
+
+        if (me->HasUnitState(UNIT_STATE_CASTING))
+            return;
+
+        events.Update(diff);
+
+        while (uint32 eventId = events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_RAYS_OF_ANGUISH:
```

Comment:
Use ExecuteEvent hook from BossAI instead this big switch, is more readable.

Reply:
no, a switch is more readable, and the reason we use it is to use while, which makes a difference in tick updates when executing 👀